core/controllers/admin.py

```diff
@@ -855,3 +855,15 @@ def post(self):
             self.render_json({})
         else:
             raise self.InvalidInputException('This app cannot send emails.')
+
+
+class UpdateUsernameHandler(base.BaseHandler):
+    """Handler for renaming usernames."""
+
+    @acl_decorators.can_access_admin_page
+    def put(self):
+        current_username = self.payload.get('current_username', None)
+        new_username = self.payload.get('new_username', None)
+        user_id = user_services.get_user_id_from_username(current_username)
+        user_services.set_username(user_id, new_username)
```

Comment:
@seanlip Do we need to keep a record of these changes (UserSettings is not a versioned model)? Also, I found this (image below) comment in set_username method, do you know what/why we need to make these changes?  
![image](https://user-images.githubusercontent.com/16653571/80657736-0fa49200-8aa2-11ea-927c-50e20cd70a66.png)

Reply:
Let's add a write-only model in core.storage.audit which keeps track of these changes, that's a good idea.

For the comment, I think the idea was to add the old name to the "banned" names property to prevent reuse. (Maybe @iamprayush you could edit the comment to make it more clear?) But in this case I think it's not needed because the old usernames are too long anyway.

Reply:
@seanlip, I checked, and there's a note in config_domain.py saying that banned_usernames is one of the properties that has been deprecated and should not be used anymore. Should I remove the comment in set_username?

Also, does the model for tracking username changes need to be added in this PR or a separate one?

Reply:
Oh, yeah, we changed to a roles system, I forgot. Yes, let's just drop the comment then. Thanks!

Can you add the model in this PR? Should be straightforward, but let me know if not. I think we should bundle it just for safety so that we can revert any errors if needed.

Reply:
@seanlip, I was adding the versioned UsernameModel, and you've suggested to add it in core.storage.audit, but shouldn't it be added in core.storage.user?

Reply:
Sorry, I meant a write-only model in core.storage.audit, similar to the existing RoleQueryAudit model over there, which only keeps a record of username changes. You don't need to make a separate versioned model to store usernames for retrieval. This is just for auditing.

Reply:
Added the model.